portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -81,15 +106,61 @@ private String convertListToJson(List list) {
     return gson.toJson(list);
   }
 
+  /**
+   * Return a List with comments retrieved by a query
+   */
+  private List<Comment> createCommentsList(Iterable<Entity> commentsRetrieved) {
+    List<Comment> comments = new ArrayList<>();
+    for (Entity commentEntity : commentsRetrieved) {
+      long id = commentEntity.getKey().getId();
+      String username = (String) commentEntity.getProperty("username");
+      String comment = (String) commentEntity.getProperty("comment");
+      Date date = (Date) commentEntity.getProperty("date");
+
+      comments.add(new Comment(id, username, comment, date));
+    }
+
+    return comments;
+  }
+  
   /**
    * Return the request parameter or the default value if the parameter
    * was not specified by the client
    */
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
-    if (value.isEmpty()) {
+    if (value == null || value.isEmpty()) {
       return defaultValue;
     }
     return value;
   }
+
+  /**
+   * Get and assign parameters of a request
+   */
+  private void handleGetParams(HttpServletRequest request) {
+    String quantity = getParameter(request, "quantity", "");
+    if (quantity.isEmpty()) {
+      fetchQuantity = 5;
+    } else {
+      fetchQuantity = Integer.parseInt(quantity);
+    }
+
+    String dateOrder = getParameter(request, "dateOrder", "");
+    if (dateOrder.isEmpty()) {
+      isOrderByDate = false;
+    } else {
+      isOrderByDate = Boolean.parseBoolean(dateOrder);
+    }
+  }
+
+  /**
+   * Return the results of a query with a limit
+   */
+  private Iterable<Entity> retrieveComments(Query fetchComments, int quantity) {
+    if (quantity < 0) {quantity = 0;}
```

Comment:
I think you can improve readability here, it is not a problem with one statement, but if you have multiple of these lines, it may not be as readable as this example:
if (quantity < 0) {
  quantity = 0;
}

Reply:
Will do! Thanks.